src/include/processor/operator/gds_call_shared_state.h

```diff
@@ -134,6 +132,8 @@ struct KUZU_API GDSCallSharedState {
     std::unique_ptr<NodeOffsetMaskMap> inputNodeMask = nullptr;
     std::unique_ptr<NodeOffsetMaskMap> outputNodeMask = nullptr;
     std::unique_ptr<NodeOffsetMaskMap> pathNodeMask = nullptr;
+    std::unique_ptr<NodeOffsetMaskMap> graphNodeMask = nullptr;
```

Comment:
Would there ever be a case we have both `inputNodeMask` and `graphNodeMask` set?

Reply:
Also let's discuss whether we should move the ownership of `graphNodeMask` in `Graph/OnDiskGraph`. I think the latter one makes more sense.

Reply:
No so we are separating the recursive join and GDS.